win32ss/gdi/ntgdi/freetype.c

```diff
@@ -705,14 +705,28 @@ LONG FASTCALL IntWidthMatrix(FT_Face face, FT_Matrix *pmat, LONG lfWidth)
     return lfWidth;
 }
 
+FT_Fixed FASTCALL FT_FixedFromFLOATOBJ(const FLOATOBJ *pef)
+{
+    FLOATOBJ efTemp;
+    if (FLOATOBJ_Equal1(pef))
+    {
+        return INT_TO_FIXED(1);
+    }
+
+    efTemp = *pef;
+    FLOATOBJ_MulLong(&efTemp, INT_TO_FIXED(1));
```

Comment:
i don't understand that, can you explain to me?

Reply:
`INT_TO_FIXED` is defined in `$(FREETYPE)\include\freetype\internal\ftcalc.h` as follows:
```c
#define INT_TO_FIXED( x )      ( (FT_Long)(x) * 65536 )  /* << 16 */
```
Thus this macro makes an integer value a 16.16 fixed point (`FT_Fixed`). `FIXED` means `FT_Fixed` in FreeType. This code is multiplication of a `FLOATOBJ` value and a `FT_Fixed` value. You can see `FT_Matrix` in `$(FREETYPE)\include\freetype\fttypes.h`, that has its `FT_Fixed` members. `FLOATOBJ_GetLong` will return the integral part of a value multiplied by `65536`.